src/pip/_internal/commands/download.py

```diff
@@ -125,7 +125,7 @@ def run(self, options, args):
             )
 
             preparer = self.make_requirement_preparer(
-                temp_directory=directory,
+                build_dir=directory.path,
```

Comment:
The purpose of `make_requirement_preparer()` was to move the common code to a central place, and this PR seems to be undoing some of that. The helper here is for a specific use case where we will know what the `TempDirectory` is rather than a general-purpose helper.

Reply:
Was this PR for some other purpose? Maybe there is another way to achieve that purpose..

Reply:
OK, that's fair - there didn't seem a point to having `make_requirement_preparer` know about `TempDirectory`, but not having to put `.path` in several places is reasonable. I'm planning to try simplifying `TempDirectiry` (to always create at initialization, which we currently do in almost all cases), this was just removing what seemed like an unnecessary use prior to that.

Reply:
OK to keep `build_dir` and assert that `.path` is not `None`, in your opinion?

Reply:
Yeah, that's what I was thinking, though I think `temp_build_dir` would be a better name to reflect that it's a `TempDirectory` object. One thing I'm confused about is why, if `.path` can be `None`, did the type checker not complain since you had the argument annotated as `str` and you didn't have an assertion at the call site saying it wasn't `None` (so why would an assertion now be needed inside `make_requirement_preparer()`).

Reply:
`reveal_type` indicates that mypy thinks `TempDirectory.path` is `Any`. Avoiding having to put asserts everywhere was one of the motivations for trying to move creation to init.

Reply:
Oh, if this is our `TempDirectory` implementation, then we should be able to add annotations there to help with this. (Using `cast()` would be another option -- we have a dummy implementation in `misc.py`, I believe. But since the implementation is under our control, we probably shouldn't need to.)